packages/swingset-liveslots/src/facetiousness.js

```diff
@@ -41,3 +43,43 @@ export function assessFacetiousness(obj) {
   // empty objects are methodless Far objects
   return /** @type {Facetiousness} */ (result || 'one');
 }
+
+export const checkAndUpdateFacetiousness = (tag, desc, proposedFacetNames) => {
+  // The first time a durable kind gets a definition, the saved
+  // descriptor will have neither ".unfaceted" nor ".facets", and we
+  // must update the details in the descriptor. When a later
+  // incarnation redefines the behavior, we must check for
+  // compatibility (all old facets must still be defined).
+
+  if (desc.unfaceted && proposedFacetNames) {
+    Fail`defineDurableKindMulti called for unfaceted KindHandle ${tag}`;
+  }
+  if (desc.facets && !proposedFacetNames) {
+    Fail`defineDurableKind called for faceted KindHandle ${tag}`;
+  }
+  let newFacetNames;
+  if (proposedFacetNames) {
+    const oldFacetNames = desc.facets ? [...desc.facets] : [];
+    const proposal = [...proposedFacetNames];
+    newFacetNames = [];
+    // all old facets must be present in the proposal
+    for (const facet of oldFacetNames) {
+      const proposedIdx = proposal.indexOf(facet);
+      if (proposedIdx === -1) {
+        const orig = oldFacetNames.join(',');
+        const newer = proposedFacetNames.join(',');
+        Fail`durable kind "${tag}" facets (${newer}) is missing ${facet} from original definition (${orig})`;
+      }
+      proposal.splice(proposedIdx, 1); // remove from proposal
+      newFacetNames.push(facet);
+    }
+
+    // new facets are appended in alphabetic order
+    proposal.sort();
+    desc.facets = newFacetNames.concat(proposal);
+  } else {
+    desc.unfaceted = true;
+  }
```

Comment:
What is this `else` clause doing?  Isn't it redundant with the check on line 54?

Reply:
Line 81 is where `.unfaceted` is assigned in the first place. `checkAndUpdateFacetiousness` is called on both the first and the subsequent times. The first time it gets a descriptor with just `{ kindID, tag }`, and this function is responsible for adding `.unfaceted` or `.facets` (it mutates `desc` in place.. I should call that out in a comment). The subsequent times, it might update `.facets` (if new facets were added), redundantly re-assign `.facets` with the same names as before (if no new facets were added), or it might redundantly re-assign `.unfaceted = true`.

It seemed easier to have it always make an assignment, even if it's reassigning the same value, than to have a bunch of conditionals.

I'll add some comments.